### PR TITLE
make EnhancedDoubleHasher.mod() public

### PR DESCRIPTION
**Pull request by Claudenw**
Fixed COLLECTIONS-840

### CODE REVIEW

**Comment by aherbert** [comment]
-1 to simply exposing the internal method. Let's put it somewhere more logical in the package hierarchy, e.g. BitMap.

**Review by garydgregory — CHANGES_REQUESTED** [review]
Hi @Claudenw 
Would you please add a unit test in `BitMapTest` for the new `mod()` public API? Make sure to establish expected behavior for normal and edge cases, especially for what should happen if overflow or underflow takes place.
TY!

File: src/test/java/org/apache/commons/collections4/bloomfilter/EnhancedDoubleHasherTest.java [hunk 1]
```diff
@@ -99,7 +99,7 @@ void testModEdgeCases() {
         for (final long dividend : new long[] {-1, -2, -3, -6378683, -23567468136887892L, Long.MIN_VALUE, 345, 678686,
             67868768686878924L, Long.MAX_VALUE}) {
             for (final int divisor : new int[] {1, 2, 3, 5, 13, Integer.MAX_VALUE}) {
-                assertEquals((int) Long.remainderUnsigned(dividend, divisor), EnhancedDoubleHasher.mod(dividend, divisor),
+                assertEquals((int) Long.remainderUnsigned(dividend, divisor), BitMap.mod(dividend, divisor),
```
Comment (aherbert) [inline src/test/java/org/apache/commons/collections4/bloomfilter/EnhancedDoubleHasherTest.java:102]: @garydgregory I think your issue would be solved if @Claudenw moves this test to the BitMapTest. It is no longer applicable to the EnhancedDoubleHasher.

Comment (aherbert) [reply src/test/java/org/apache/commons/collections4/bloomfilter/EnhancedDoubleHasherTest.java:102]: When doing this can you add an extra case for `dividend=0`. This is not currently covered.

I don't think we need edge cases to show behaviour when the divisor is negative. This method is written for performance and omits checks for bad arguments. However the javadoc can be updated to make it a user beware method. It should add strong wording that the result is undefined if the divisor is negative, and there will be an java.lang.ArithmeticException if the divisor is zero.
